Properly fail on empty repoquery version checks

mtnbikenc opened the pull request:
The `repoquery_cmd` could return an empty string if RPMs are not available.  This PR adds a task to confirm the `repoquery_cmd` returned a result, or fail with a helpful error message.

Fixes #3119

mtnbikenc commented:
aos-ci-test

ingvagabund commented on playbooks/common/openshift-cluster/upgrades/pre/verify_upgrade_targets.yml:
```diff
@@ -31,6 +31,13 @@
     register: avail_openshift_version
     when: not openshift.common.is_containerized | bool
 
+  - name: Fail when unable to determine available OpenShift RPM version
```
If you move `set_fact` after `Verify containers are available for upgrade`, you can put:
* `set_fact`
* `Check latest available OpenShift RPM version`
* `Fail when unable to determine available OpenShift RPM version`
* `Verify OpenShift RPMs are available for upgrade`

under common block with one `not openshift.common.is_containerized | bool` when condition.

mtnbikenc replied:
I had made a bunch of changes to this file to standardize stuff, including putting in blocks, but then backed it all out because I didn't want to conflate the actual change I was making to put in the one `fail` task.  I'll go ahead and make the other standardization changes.

mtnbikenc commented:
@ingvagabund @tbielawa Updated this PR with additional changes as follows:
* Used `block` to group common `when` conditions
* Removed unused/unnecessary vars
* Added names to all tasks
* Converted `when` statement conditionals to lists

mtnbikenc commented:
aos-ci-test

ingvagabund commented:
aos-ci-test

mtnbikenc commented:
aos-ci-test

sdodson commented:
[merge]

mtnbikenc commented:
[merge]
